### PR TITLE
feat(resilience): add debtSustainabilityGap indicator (IMF DSA construct)

### PR DESCRIPTION
**Pull request by koala73**
## Summary

- Adds a fourth `fiscalSpace` indicator — `debtSustainabilityGap` — capturing the IMF DSA construct `gap = pb − ((r−g)/(1+g))·d`. Positive = debt path declining, negative = rising.
- Computed at seed time (not score time) from 5 already-seeded IMF WEO series; scorer just normalizes the precomputed field.
- Rebalances `fiscalSpace` weights from `0.4/0.3/0.3` (3 indicators) to `0.25/0.20/0.20/0.35` (4 indicators). Gap becomes the largest single slice — the only indicator that integrates pb, r, g, d, and their interaction term.

## Why

The current `fiscalSpace` dimension blends revenue / fiscal balance / debt level. None of those — alone or blended — actually answers "is this country's debt path sustainable?" A country can have high debt + strong primary surplus + g≈r and be fine (Japan-ish), or low debt + chronic primary deficit + r>g and be in trouble. The composite penalizes the level and rewards the surplus, then averages, losing the r-g interaction term. The gap is the standard IMF DSA construct used by Article IV missions, ECB MIP scoreboard, and S&P sovereign methodology.

Plan #2 of the fiscal-resilience initiative. Plan #1 (orphan-fiscal-data UI surface) shipped in PR #3668; together they close the loop on "data we ingest but don't act on."

## Formula

```
g    = (1 + realGdpGrowth/100) × (1 + cpiInflation/100) − 1     compounded nominal growth
r    = max(0, (primaryBalance − fiscalBalance) / debtToGdp)     effective interest rate (DSA identity)
pb*  = ((r − g) / (1 + g)) × debtToGdp                          debt-stabilizing primary balance
gap  = primaryBalance − pb*                                      positive ⇒ debt declining
```

`r` is derived from the algebraic identity (overall balance = primary balance − interest expense), so `interest %GDP = primaryBalance − fiscalBalance`. Avoids needing per-country sovereign yields (FRED only covers US).

### Worked examples (end-to-end, verified in `tests/resilience-debt-sustainability-gap.test.mts`)

| Country 2024 | d | pb | fb | real g | infl | gap | normalized score |
|---|---|---|---|---|---|---|---|
| France | 110 | −2.5 | −4.5 | 0.7 | 2.1 | **−1.43** | 44.6 |
| Japan | 250 | 0 | −3 | 0.5 | 2.0 | **+3.20** | 100 (clamps) |
| Italy | 137 | +1.0 | −3.5 | 0.5 | 2.0 | **−0.04** | 62.0 |
| Norway | 38 | +10 | +8 | 1.0 | 2.5 | **+9.36** | 100 (clamps) |
| Argentina | 90 | −2 | −5 | −3.0 | 200 | inflation > 25% cap | excluded |

## Architecture

- **Seed-time computation**: the gap is computed inside `scripts/seed-recovery-fiscal-space.mjs` and stored as `debtSustainabilityGapPct` per country in the canonical Redis blob (`resilience:recovery:fiscal-space:v1`). Scorer reads the precomputed field — matches every other resilience scorer's single-blob pattern.
- **Pure exported helpers**: `latestCommonYear`, `computeDebtSustainabilityGap`, `buildFiscalSpaceCountries`, `validateFiscalSpace` are exported so tests skip ESM mocking. `FISCAL_SPACE_VALIDATION_FLOORS` and `INFLATION_GAP_CAP_PCT` exported as named constants.
- **Year alignment**: gap inputs read at `latestCommonYear` across the 5 formula series only (revenue NOT in alignment check — it's not in the formula). Year-mismatched countries → `gap=null`; fiscal-3 still scores them.
- **Country inclusion**: stays anchored to fiscal-3 union (revenue ∪ balance ∪ debt). Existing skip guard preserved verbatim — an IMF fiscal-series outage cannot emit phantom countries with null fiscal-3.
- **Two-floor validate**: payload rejected if `<150` countries have fiscal-3 OR `<100` countries have gap inputs. Either floor failing → previous canonical blob keeps serving (`_seed-utils.mjs:1010-1017` strict-floor behavior).
- **Tier classification**: `enrichment` (not Core), because the inflation cap excludes ~5-10 hyperinflation countries by design — structurally cannot meet the Core ≥180 coverage invariant. Scorer doesn't differentiate by tier; this is a quality classification only.

## Policy decisions (locked 2026-05-12)

1. Goalposts: `worst=-5 / best=+3` (DSA distress envelope)
2. Inflation cap: 25% (excludes Argentina/Lebanon/Venezuela from gap; fiscal-3 still scores them)
3. Weights: 0.25/0.20/0.20/0.35 (gap as largest single slice)
4. Backtest acceptance: ≤10 countries shifting >±5 ranks in economy pillar

## Test plan

- [x] `npx tsc --noEmit` clean
- [x] `npx biome lint` clean on touched files (8 modified + 3 new)
- [x] **750/750 resilience tests pass** (full suite)
  - 12 formula unit tests (worked examples + edge cases)
  - 20 seeder contract tests (helpers + 6-country fixture + two-floor validate)
  - 7 scorer integration tests (5-country scenarios + null redistribution)
  - 89 dimension-scorer baseline tests (existing parity)
  - 36 indicator-registry + doc-parity tests
- [x] Production seeder ran end-to-end: 191 records, both floor checks passed
- [x] Backtest comparison: Spearman 0.9731, max country drift 9.63pt, cohort median 3.17pt (all within thresholds)
- [x] Baseline updates documented in `tests/resilience-scorers.test.mts` with plan-002 → plan-2026-05-12 trace

## Backtest verdict (acceptance criteria)

| Gate | Status | Result | Threshold |
|---|---|---|---|
| Spearman rank correlation | ✅ PASS | 0.9731 | floor 0.85 |
| Max country drift | ✅ PASS | 9.63pt | ceiling 15 |
| Cohort median shift | ✅ PASS | worst 3.17pt | ceiling 10 |
| Universe integrity | ✅ PASS | 93 endpoints | — |
| Core indicator measurability | ✅ PASS | 47/51 | floor 80% |
| Matched-pair gaps | ⚠️ PRE-EXISTING | 2 failures | — |

**Plan's stated acceptance criterion ("≤10 countries shifting >±5 ranks") is satisfied.**

### Known concern — matched-pair gate (pre-existing, not introduced by this PR)

Two matched pairs were ALREADY failing on `main` before this PR:
- `fr-vs-de`: currentGap=−5.19 (failing) → proposedGap=−6.70 (amplified by 3.79pt)
- `sg-vs-ch`: currentGap=−13.31 (failing) → proposedGap=−16.51 (amplified by 3.20pt)

My change amplifies the pre-existing miss but does not introduce a new directional failure. The amplification is structurally explainable: France's high-debt + mild-deficit profile takes a small additional gap-indicator penalty. Recommend treating as separate tech-debt follow-up — the matched-pair calibration predates this PR.

### Pre-merge (done)
- All tests pass (750/750)
- Local seeder dry-run successful (191 records)
- Formula worked examples pinned in unit tests ±0.01 tolerance

### Post-merge / post-deploy
- Wait **1-2 seeder cron ticks** (the cron interval is monthly per `seed-bundle-resilience-recovery.mjs:5`, so the honest verification window is 30-60 days after merge — first tick may hiccup, second is the backstop). Direct Redis GET on `resilience:recovery:fiscal-space:v1` and assert ≥100 countries have `debtSustainabilityGapPct: number`. Spot-check France/Japan/Norway against worked examples ±0.5 (tolerance accommodates IMF data refresh).
- Run `scripts/compare-resilience-current-vs-proposed.mjs` against pre/post Redis snapshots to confirm the expected directional shifts (Italy/Greece drop, Japan rises, Norway/Singapore rise, Argentina/Lebanon/Venezuela excluded from gap signal but fiscal-3 unchanged).
- `/api/health` shows seeder freshness — useful, but not field-level coverage. Don't rely on health for field verification.

### Rollback
- Revert the PR. Seeder `schemaVersion` drops back to 1 on next tick. Old scorer reads the new blob fine — extra fields are ignored, the 3 fiscal-3 indicators score returns. Zero data loss; a few hours of degraded scoring between revert and next seeder tick.

## Plan provenance

`plans/add-debt-sustainability-gap-indicator.md` — approved by Codex after 4 rounds of review on 2026-05-12. Implementation review by Codex on the actual diff also completed before merge. Plan-doc is included in this PR for design-rationale provenance.

### CODE REVIEW

File: tests/seed-recovery-fiscal-space.test.mts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  it('returns the most recent year present in every map', () => {
+    const y = latestCommonYear([
+      byYear(['2024', 110], ['2023', 105]),
+      byYear(['2024', -2.5], ['2023', -2.0]),
+      byYear(['2024', -4.5]),
+      byYear(['2024', 0.7]),
+      byYear(['2024', 2.1]),
+    ]);
+    assert.equal(y, 2024);
+  });
```
[the remark this answers is not in the document]
Comment (koala73) [reply tests/seed-recovery-fiscal-space.test.mts:34]: Addressed in commit `9401d7031` — all 3 `latestCommonYear` tests now derive years from `new Date().getFullYear()`. Caught two additional silent-degradation cases (lines 50-58 and 60-63) while fixing the explicit time-bomb you flagged at line 28: those don't outright break in 2027 but they stop testing what they claim to test (returning null because years are out of scope rather than because there's no overlap / NaN). All 3 now exercise the intended branches in every future year.

File: tests/resilience-debt-sustainability-gap.test.mts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    assert.ok(Math.abs(gap! - 9.36) < 0.05, `expected ≈ +9.36, got ${gap}`);
+  });
+
+  it('Italy 2024: high-debt, mild surplus → gap near zero (debt-stabilizing)', () => {
+    const gap = computeDebtSustainabilityGap({
+      debt: 137, pb: 1.0, fb: -3.5, realG: 0.5, infl: 2.0,
+    });
+    assert.ok(gap !== null, 'gap should be defined for Italy inputs');
+    // Plan example: gap ≈ -0.04 (right at the stabilizing point)
+    assert.ok(Math.abs(gap!) < 0.5, `expected gap near 0 for stabilizing Italy, got ${gap}`);
```
[the remark this answers is not in the document]
Comment (koala73) [reply tests/resilience-debt-sustainability-gap.test.mts]: Addressed in commit `9401d7031` — Italy tolerance tightened from `< 0.5` to `< -0.04, 0.01` matching the other 4 worked-example assertions. The wider tolerance would have silently passed any formula regression producing values in (-0.5, +0.5).
